src/datanode/src/store.rs

```diff
@@ -58,16 +58,31 @@ pub(crate) async fn new_object_store(opts: &DatanodeOptions) -> Result<ObjectSto
         object_store
     };
 
-    Ok(object_store
-        .layer(MetricsLayer)
+    let store = object_store
         .layer(
             LoggingLayer::default()
                 // Print the expected error only in DEBUG level.
                 // See https://docs.rs/opendal/latest/opendal/layers/struct.LoggingLayer.html#method.with_error_level
                 .with_error_level(Some("debug"))
                 .expect("input error level must be valid"),
         )
-        .layer(TracingLayer))
+        .layer(TracingLayer);
+
+    // In the test environment, multiple datanodes will be started in the same process.
+    // If each datanode registers Prometheus metric when it starts, it will cause the program to crash. (Because the same metric is registered repeatedly.)
+    // So the Prometheus metric layer is disabled in the test environment.
+    #[cfg(feature = "testing")]
+    return Ok(store);
+
+    #[cfg(not(feature = "testing"))]
+    {
+        let registry = prometheus::default_registry();
+        Ok(
+            store.layer(object_store::layers::PrometheusLayer::with_registry(
+                registry.clone(),
+            )),
```

Comment:
question: Will it panic if we create multiple object stores? Maybe we should implement a new layer.

Reply:
It will. And those feature gates are added to avoid panicking during tests